doc/tutorials/basic/build_chatbot.md

```diff
@@ -0,0 +1,96 @@
+# Build a Chat Bot
+
+In this tutorial, we will build a streaming *chat bot*. We will first use the *high-level* [`ChatInterface`](../../reference/chat/ChatInterface.ipynb) to build a basic chat bot. Then we will add streaming.
+
+:::{note}
+When we ask to *run the code* in the sections below, we may either execute the code directly in the Panel docs via the green *run* button, in a cell in a notebook, or in a file `app.py` that is served with `panel serve app.py --autoreload`.
+:::
+
+## Build a Basic Chat Bot
+
+Run the code below:
+
+```{pyodide}
+import panel as pn
+from time import sleep
+
+pn.extension()
+
+def get_response(contents, user, instance):
+    if "turbine" in contents.lower():
+        response = "A wind turbine converts wind energy into electricity."
+    else:
+        response = "Sorry, I don't know."
+    sleep(1)
+    return response
+
+chat_bot = pn.chat.ChatInterface(callback=get_response, max_height=500)
+chat_bot.send("Ask me what a wind turbine is", user="Assistant", respond=False)
+chat_bot.servable()
+```
+
+Try entering `What is a wind turbine?` in the *text input* and click *Send*.
+
+:::{note}
+The `callback` function `get_response` will receive `"What is a wind turbine?"` in the `contents` argument. Since `contents` contains the word *turbine*, the chat bot will return `"A wind turbine converts wind energy into electricity."` as the `response`.
+:::
+
+## Add Streaming
+
+We will now make the chat bot *stream* its response just like ChatGPT does.
+
+Run the code below:
+
+```{pyodide}
+import panel as pn
+from time import sleep
+
+pn.extension()
+
+def get_response(contents, user, instance):
+    if "turbine" in contents.lower():
+        response = "A wind turbine converts wind energy into electricity."
+    else:
+        response = "Sorry, I don't know."
+    for index in range(len(response)):
+        yield response[0:index+1]
+        sleep(0.03)
+
+chat_bot = pn.chat.ChatInterface(callback=get_response, max_height=500)
+chat_bot.send("Ask me anything!", user="Assistant", respond=False)
+chat_bot.servable()
+```
+
+Try entering `What is a wind turbine?` in the *text input* and click *Send*.
+
+:::{note}
+The chat app is now *streaming* because we `yield` the partial `response` instead of `return`ing the full `response`.
+:::
+
+:::{note}
+To make the Streaming Chat Bot scale to many users you should be using `async`. You will learn more about scaling applications and `async` in the [Intermediate Tutorials](../intermediate/index.md).
```

Comment:
If you want to support many users `sync` is an antipattern. Just as using `.show` instead of `panel serve`. But single user/ few user use cases also have their place. That is where a lot of people start. Or its what they use for exploration.

I think its ok. @ahuang11 . Any recommendations on this comment?

Reply:
This is why I believe building a chat app would be worth its own tutorial, to show and explain this sort of thing.

Reply:
But this is that tutorial. If it needs more explanation or detail any one can make a PR to make it better.

Reply:
What I was trying to say is that I believe there should only be one "build a chat app tutorial" and not a basic one, an intermediate one, an advanced one, etc. We have to make decisions, that's hard, but that's better for the user in the end.

See it like the problem we have with APIs. We have too many of them that do the same-ish thing. Let's not apply the same scheme for the tutorials, users will have the How-Tos, Reference and Explanations to learn more and find out about the rest of things you can do.

Reply:
I agree that this is that tutorial, whether it lives in basic or gets it's own entirely separate top-level section is up in the air I think but we did already decide to separate this out from the main flow of the basic tutorial. In the long run I do aim to have multiple sections for the chat app tutorial but for now this seems totally fine.